databricks/koalas/window.py

```diff
@@ -42,6 +60,369 @@ def __getattr__(self, item: str) -> Any:
+        Note that the given `func` name should be same as the API's method name.
+        """
+        from databricks.koalas import DataFrame, Series
+
+        if isinstance(self.kdf_or_kser, Series):
+            kser = self.kdf_or_kser
+            return kser._with_new_scol(
+                func(kser._scol)).rename(kser.name)
+        elif isinstance(self.kdf_or_kser, DataFrame):
+            kdf = self.kdf_or_kser
+            applied = []
+            for column in kdf.columns:
+                applied.append(
+                    getattr(kdf[column].rolling(self._window_val + 1,
+                            self._min_periods), func.__name__)())
+
+            sdf = kdf._sdf.select(
+                kdf._internal.index_scols + [c._scol for c in applied])
+            internal = kdf._internal.copy(
+                sdf=sdf,
+                data_columns=[c._internal.data_columns[0] for c in applied],
+                column_index=[c._internal.column_index[0] for c in applied])
+            return DataFrame(internal)
+
+    def sum(self):
+        """
+        Calculate rolling sum of given DataFrame or Series.
+
+        .. note:: the current implementation of this API uses Spark's Window without
+            specifying partition specification. This leads to move all data into
+            single partition in single machine and could cause serious
+            performance degradation. Avoid this method against very large dataset.
+
+        Returns
+        -------
+        Series or DataFrame
+            Same type as the input, with the same index, containing the
+            rolling sum.
+
+        See Also
+        --------
+        Series.sum : Reducing sum for Series.
+        DataFrame.sum : Reducing sum for DataFrame.
+
+        Examples
+        --------
+        >>> s = ks.Series([4, 3, 5, 2, 6])
+        >>> s
+        0    4
+        1    3
+        2    5
+        3    2
+        4    6
+        Name: 0, dtype: int64
+
+        >>> s.rolling(2).sum()
+        0    NaN
+        1    7.0
+        2    8.0
+        3    7.0
+        4    8.0
+        Name: 0, dtype: float64
+
+        >>> s.rolling(3).sum()
+        0     NaN
+        1     NaN
+        2    12.0
+        3    10.0
+        4    13.0
+        Name: 0, dtype: float64
+
+        For DataFrame, each rolling max is computed column-wise.
+
+        >>> df = ks.DataFrame({"A": s.to_numpy(), "B": s.to_numpy() ** 2})
+        >>> df
+           A   B
+        0  4  16
+        1  3   9
+        2  5  25
+        3  2   4
+        4  6  36
+
+        >>> df.rolling(2).sum()
+             A     B
+        0  NaN   NaN
+        1  7.0  25.0
+        2  8.0  34.0
+        3  7.0  29.0
+        4  8.0  40.0
+
+        >>> df.rolling(3).sum()
+              A     B
+        0   NaN   NaN
+        1   NaN   NaN
+        2  12.0  50.0
+        3  10.0  38.0
+        4  13.0  65.0
+        """
+        def sum(scol):
+            window = Window.orderBy(self._index_scols)
```

Comment:
Can we move this into `_apply_as_series_or_frame` and use `self._window`? We can match the implementation into Expanding one I think.

Reply:
is it possible although this `window` and `self._window` are used for different purpose like below?

<img width="590" alt="스크린샷 2019-11-06 오후 3 18 34" src="https://user-images.githubusercontent.com/44108233/68273037-bcc8bf80-00a8-11ea-96bb-0c6b69bd1167.png">